packages/protocol/scripts/truffle/make-release.ts

```diff
@@ -0,0 +1,219 @@
+    }
+  }
+
+  public set = (contract: string, address: Address) => {
+    this.addresses.set(contract, address)
+  }
+}
+
+module.exports = async (callback: (error?: any) => number) => {
+  try {
+    const argv = require('minimist')(process.argv.slice(2), {
+      string: ['report', 'network', 'proposal', 'libraries', 'initialize_data', 'build_directory'],
+    })
+    const report: ASTDetailedVersionedReport = readJsonSync(argv.report).report
+    const initializationData = readJsonSync(argv.initialize_data)
+    const dependencies = new ContractDependencies(linkedLibraries)
+    const contracts = readdirSync(join(argv.build_directory, 'contracts')).map((x) =>
+      basename(x, '.json')
+    )
+    const registry = await artifacts
+      .require('Registry')
+      .at('0x000000000000000000000000000000000000ce10')
+    const addresses = await ContractAddresses.create(contracts, registry)
+    const released: Set<string> = new Set([])
+    const proposal = []
+    // Release 1 will deploy all libraries with proxies so that they're more easily
+    // upgradable. All contracts that link libraries should be upgraded to instead link the proxied
+    // library.
+    // To ensure this actually happens, we check that all contracts that link libraries are marked
+    // as needing to be redeployed.
+    // TODO(asa): Remove this check after release 1.
+    const linksLibrariesWithoutChanges = contracts.map(
+      (c) =>
+        dependencies.get(c).length &&
+        !Object.keys(report.contracts).includes(c) &&
+        !c.endsWith('Test')
+    )
+    if (linksLibrariesWithoutChanges.some((b) => b)) {
+      contracts.forEach((c, i) => {
+        if (linksLibrariesWithoutChanges[i]) {
+          console.log(
+            `${c} links ${dependencies.get(c)} and needs to be upgraded to link proxied libraries.`
+          )
+        }
+      })
+      throw new Error('All contracts linking libraries should be upgraded in release 1')
+    }
+
+    const release = async (contractName: string) => {
+      if (released.has(contractName)) {
+        return
+      } else {
+        // 1. Release all dependencies.
+        const contractDependencies = dependencies.get(contractName)
+        for (const dependency of contractDependencies) {
+          await release(dependency)
+        }
+        // 2. Link dependencies.
+        const Contract = await artifacts.require(contractName)
+        await Promise.all(contractDependencies.map((d) => Contract.link(d, addresses.get(d))))
+
+        // This is a hack that will re-deploy all libraries with proxies, whether or not they have
+        // changes to them.
+        // TODO(asa): Remove `isLibrary` for future releases.
+        const isLibrary = Object.keys(linkedLibraries).includes(contractName)
+        const deployImplementation = Object.keys(report.contracts).includes(contractName)
+        const deployProxy =
+          deployImplementation && report.contracts[contractName].changes.storage.length
```

Comment:
Do you truly want to just be checking for the existence of the `length` property or assert a `length > 0`?

Reply:
The latter, is `0` not falsey? Changing to > 0